app/models/service_template_orchestration.rb

```diff
@@ -57,6 +57,29 @@ def self.validate_config_info(options)
 
   private
 
+  def update_service_resources(config_info, _auth_user = nil)
+    if config_info[:template_id] && config_info[:template_id] != orchestration_template.id
+      self.orchestration_template = OrchestrationTemplate.find(config_info[:template_id])
+    elsif config_info[:template] && config_info[:template] != orchestration_template
+      self.orchestration_template = config_info[:template]
+    end
+
+    if config_info[:manager_id] && config_info[:manager_id] != orchestration_manager.id
+      self.orchestration_manager = ExtManagementSystem.find(config_info[:manager_id])
+    elsif config_info[:manager] && config_info[:manager] != orchestration_manager
+      self.orchestration_manager = config_info[:manager]
+    end
+  end
+
+  def validate_update_config_info(options)
+    super
+    config_info = options[:config_info]
```

Comment:
code reuse - `self.class.validate_config_info(options)`, same for the other class.